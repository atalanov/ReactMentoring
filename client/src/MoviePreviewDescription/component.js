import React from 'react';
import { Link } from 'react-router-dom';

export default props => (
  <div>
    <h4><Link to={`/movies/${props.id}`}>{props.name}</Link></h4>
    <p><Link to={`/movies/${props.id}`}>{props.jenre} ({props.year})</Link></p>
  </div>
);

