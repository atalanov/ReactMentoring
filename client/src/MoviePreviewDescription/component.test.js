import React from 'react';
import { shallow, render } from 'enzyme';
import MoviePreviewDescription from './component';

describe('MoviePreviewDescription tests', () => {
    test('Component MoviePreviewDescription renders correctly', () => {
        const component = shallow(<MoviePreviewDescription name="TestFilm" jenre="Horror" year="2018" />);
        expect(component).toMatchSnapshot();
    });
});
