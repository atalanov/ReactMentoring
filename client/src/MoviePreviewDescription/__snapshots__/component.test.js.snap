// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MoviePreviewDescription tests Component MoviePreviewDescription renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    jenre="Horror"
    name="TestFilm"
    year="2018"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h4>
          <Link
            replace={false}
            to="/movies/undefined"
          >
            TestFilm
          </Link>
        </h4>,
        <p>
          <Link
            replace={false}
            to="/movies/undefined"
          >
            Horror
             (
            2018
            )
          </Link>
        </p>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/movies/undefined"
          >
            TestFilm
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": "TestFilm",
            "replace": false,
            "to": "/movies/undefined",
          },
          "ref": null,
          "rendered": "TestFilm",
          "type": [Function],
        },
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <Link
            replace={false}
            to="/movies/undefined"
          >
            Horror
             (
            2018
            )
          </Link>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              "Horror",
              " (",
              "2018",
              ")",
            ],
            "replace": false,
            "to": "/movies/undefined",
          },
          "ref": null,
          "rendered": Array [
            "Horror",
            " (",
            "2018",
            ")",
          ],
          "type": [Function],
        },
        "type": "p",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h4>
            <Link
              replace={false}
              to="/movies/undefined"
            >
              TestFilm
            </Link>
          </h4>,
          <p>
            <Link
              replace={false}
              to="/movies/undefined"
            >
              Horror
               (
              2018
              )
            </Link>
          </p>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/movies/undefined"
            >
              TestFilm
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": "TestFilm",
              "replace": false,
              "to": "/movies/undefined",
            },
            "ref": null,
            "rendered": "TestFilm",
            "type": [Function],
          },
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <Link
              replace={false}
              to="/movies/undefined"
            >
              Horror
               (
              2018
              )
            </Link>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": Array [
                "Horror",
                " (",
                "2018",
                ")",
              ],
              "replace": false,
              "to": "/movies/undefined",
            },
            "ref": null,
            "rendered": Array [
              "Horror",
              " (",
              "2018",
              ")",
            ],
            "type": [Function],
          },
          "type": "p",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
